batch7crud-roll4/src/main/java/com/lftechnology/batch7crud/controller/EmployeeController.java

```diff
@@ -0,0 +1,187 @@
+    String firstName = request.getParameter(PARAM_FIRST_NAME);
+    String lastName = request.getParameter(PARAM_LAST_NAME);
+    String station = request.getParameter(PARAM_STATION);
+
+    Employee employee = new Employee();
+    employee.setFirstName(firstName);
+    employee.setLastName(lastName);
+    employee.setStation(station);
+
+    try {
+      employeeService.save(employee);
+      response.sendRedirect(request.getContextPath() + ROUTE_EMPLOYEES);
+    } catch (DataException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      show500(request, response, e);
+    }
+  }
+
+  private void viewProfile(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    try {
+      int employeeId = parameterValueAsInt(request, 2);
+
+      request.setAttribute(ATTRIB_EMPLOYEE, employeeService.fetchById(employeeId));
+      RequestDispatcher view = request.getRequestDispatcher(URL_EMPLOYEE_PROFILE_PAGE);
+      view.forward(request, response);
+    } catch (DataException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      show500(request, response, e);
+    } catch (NumberFormatException e) {
+      show404(request, response);
+    }
+  }
+
+  private void edit(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    try {
+      int employeeId = parameterValueAsInt(request, 2);
+
+      request.setAttribute(ATTRIB_EMPLOYEE, employeeService.fetchById(employeeId));
+      RequestDispatcher view = request.getRequestDispatcher(URL_EMPLOYEE_EDIT_PAGE);
+      view.forward(request, response);
+    } catch (DataException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      show500(request, response, e);
+    } catch (NumberFormatException e) {
+      show404(request, response);
+    }
+
+  }
+
+  private void editProcess(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    try {
+      int employeeId = parameterValueAsInt(request, 2);
+
+      String firstName = request.getParameter(PARAM_FIRST_NAME);
+      String lastName = request.getParameter(PARAM_LAST_NAME);
+      String station = request.getParameter(PARAM_STATION);
+
+      Employee employee = new Employee();
+      employee.setId(employeeId);
+      employee.setFirstName(firstName);
+      employee.setLastName(lastName);
+      employee.setStation(station);
+
+      employeeService.update(employee);
+      response.sendRedirect(request.getContextPath() + ROUTE_EMPLOYEES);
+    } catch (DataException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      show500(request, response, e);
+    } catch (NumberFormatException e) {
+      show404(request, response);
+    }
+
+  }
+
+  private void deleteProcess(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    try {
+      int employeeId = parameterValueAsInt(request, 2);
+      employeeService.deleteEmployee(employeeId);
```

Comment:
What happens if employee is not deleted? Or if the employee with id does not exist?